common/sampling.h

```diff
@@ -36,6 +36,9 @@ typedef struct llama_sampling_params {
     float       cfg_scale     = 1.f; // how strong is guidance
 
     std::unordered_map<llama_token, float> logit_bias; // logit bias for specific tokens
+
+    std::vector<llama_token> penalty_prompt_tokens;
+    bool                     use_penalty_prompt_tokens = false;
```

Comment:
Can we avoid this `bool use_penalty_prompt_tokens` flag?
It seems it can be replaced with `!penalty_prompt_tokens.empty()`

Reply:
Not quite. These params are independent. When `penalty_prompt_tokens` is empty, `use_penalty_prompt_tokens` can still be `true`. It will mean that the `server` must use custom penalty prompt, but it's just empty at the start. Otherwise there will be no distinction between "do not use custom prompt" and "use empty custom prompt".